restcomm/restcomm.http/src/main/java/org/restcomm/connect/http/OrganizationsEndpoint.java

```diff
@@ -0,0 +1,253 @@
+                        organization = organizationsDao.getOrganization(new Sid(organizationSid));
+                    } else {
+                        return status(FORBIDDEN).build();
+                    }
+                } else {
+                    organization = organizationsDao.getOrganization(new Sid(organizationSid));
+                }
+            } catch (Exception e) {
+                return status(NOT_FOUND).build();
+            }
+        }
+
+        if (organization == null) {
+            return status(NOT_FOUND).build();
+        } else {
+            if (APPLICATION_XML_TYPE == responseType) {
+                final RestCommResponse response = new RestCommResponse(organization);
+                return ok(xstream.toXML(response), APPLICATION_XML).build();
+            } else if (APPLICATION_JSON_TYPE == responseType) {
+                return ok(gson.toJson(organization), APPLICATION_JSON).build();
+            } else {
+                return null;
+            }
+        }
+    }
+
+    /**
+     * @param info
+     * @param responseType
+     * @return
+     */
+    protected Response getOrganizations(UriInfo info, final MediaType responseType) {
+        checkAuthenticatedAccount();
+        allowOnlySuperAdmin();
+
+        List<Organization> organizations = null;
+
+        String status = info.getQueryParameters().getFirst("Status");
+
+        if(status != null && Organization.Status.getValueOf(status.toLowerCase()) != null){
+            organizations = organizationsDao.getOrganizationsByStatus(Organization.Status.getValueOf(status.toLowerCase()));
+        }else{
+            organizations = organizationsDao.getAllOrganizations();
+        }
+        if (organizations == null || organizations.isEmpty()) {
+            return status(NOT_FOUND).build();
+        }
+        if (APPLICATION_XML_TYPE == responseType) {
+            final RestCommResponse response = new RestCommResponse(new OrganizationList(organizations));
+            return ok(xstream.toXML(response), APPLICATION_XML).build();
+        } else if (APPLICATION_JSON_TYPE == responseType) {
+            return ok(gson.toJson(organizations), APPLICATION_JSON).build();
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * putOrganization create new organization
+     * @param domainName
+     * @param data
+     * @param applicationJsonType
+     * @return
+     */
+    protected Response putOrganization(String domainName, MultivaluedMap<String, String> data,
+            MediaType responseType) {
+        if(domainName == null){
+            return status(BAD_REQUEST).entity(MSG_EMPTY_DOMAIN_NAME ).build();
+        }else{
+            checkAuthenticatedAccount();
+            allowOnlySuperAdmin();
+
+            //Character verification
+            final Pattern pattern = Pattern.compile(SUB_DOMAIN_NAME_VALIDATION_PATTERN);
```

Comment:
ca nwe compile the patterns in intialization and reuse them?

Reply:
done